android/app/src/main/java/org/fossasia/openevent/viewmodels/MainActivityViewModel.java

```diff
@@ -0,0 +1,219 @@
+import org.fossasia.openevent.events.TracksDownloadEvent;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.DateConverter;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+import org.fossasia.openevent.utils.Utils;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import io.realm.Realm;
+import io.realm.RealmList;
+import io.realm.RealmResults;
+import timber.log.Timber;
+
+public class MainActivityViewModel extends ViewModel {
+    private RealmDataRepository realmRepo;
+    private LiveData<Event> eventLiveData;
+    private Event event;
+    private boolean newInstance = false;
+
+    public MainActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<Event> getEventSync() {
+        newInstance = false;
+        if (eventLiveData == null) {
+            event = realmRepo.getEventSync();
+            if (event == null) {
+                newInstance = true;
+                event = new Event();
+            }
+            LiveRealmDataObject<Event> liveRealmDataObject = RealmDataRepository.asLiveDataForObject(realmRepo.getEventSync());
+            eventLiveData = Transformations.map(liveRealmDataObject, input -> input);
+        }
+        return eventLiveData;
+    }
+
+    public void downloadPageId() {
+        if (!newInstance) {
+            //Store the facebook page name in the shared preference from the database
+            if (SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_NAME, null) == null) {
+                RealmList<SocialLink> socialLinks = event.getSocialLinks();
+                if (socialLinks != null) {
+                    RealmResults<SocialLink> facebookPage = socialLinks.where().equalTo("name", "Facebook").findAll();
+                    if (facebookPage.size() == 0)
+                        return;
+
+                    SocialLink facebookLink = facebookPage.get(0);
+                    String link = facebookLink.getLink();
+                    String tempString = ".com";
+                    String pageName = link.substring(link.indexOf(tempString) + tempString.length()).replace("/", "");
+
+                    if (Utils.isEmpty(pageName))
+                        return;
+
+                    SharedPreferencesUtil.putString(ConstantStrings.FACEBOOK_PAGE_NAME, pageName);
+                }
+            }
+        }
+        if (SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_ID, null) == null &&
+                SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_NAME, null) != null) {
+            APIClient.getFacebookGraphAPI().getPageId(SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_NAME, null),
+                    String.valueOf((R.string.facebook_access_token)))
```

Comment:
**!!IMPORTANT!!** 

Do you know what **`String.valueOf((R.string.facebook_access_token))` is?

It is **"2131689586"**

And it will change everytime you hit the build button

What we want is API key -> Something of this form "17**********************uo"